src/OrchardCore/OrchardCore.Queries.Core/Models/Query.cs

```diff
@@ -0,0 +1,37 @@
+using System.Text.Json.Serialization;
+using OrchardCore.ContentManagement;
+using OrchardCore.Entities;
+
+namespace OrchardCore.Queries;
+
+/// <summary>
+/// Represents a query.
+/// </summary>
+public class Query : Entity
+{
+    /// <summary>
+    /// Gets or sets the technical name of the query.
+    /// </summary>
+    public string Name { get; set; }
+
+    /// <summary>
+    /// Gets the name of the source for this query.
+    /// </summary>
+    public string Source { get; set; }
```

Comment:
Since it's not the actual `IQuerySouce` (the name left me wondering when looking at the code here, and had to open files in VS to see the actual types):

```suggestion
    public string SourceName { get; set; }
```

Reply:
`Source` it what is currently called. If we change this, I'll have to add migration support for this too. I don't think this change worth the effort. We have to update the QueryManager and Migration helper to support `Source` and `SourceName`